sql/parser/eval.go

```diff
@@ -134,15 +137,23 @@ var binOps = map[binArgs]binOp{
 		},
 	},
 	binArgs{Plus, dateType, intervalType}: {
-		returnType: DummyTimestamp,
+		returnType: DummyDate,
 		fn: func(left Datum, right Datum) (Datum, error) {
-			return DTimestamp{Time: left.(DDate).Add(right.(DInterval).Duration)}, nil
+			days, err := inDays(right.(DInterval).Duration)
+			if err != nil {
+				return DummyDate, err
+			}
+			return DDate(left.(DDate) + days), nil
```

Comment:
I've modified it to only support operations between dates, and between a date and an integer. I've taken out arithmetic between date and timestamp and between date and interval. A user will need to explicitly cast a date to a timestamp to do that.